pysoa/client/client.py

```diff
@@ -54,11 +174,153 @@ def __str__(self):
 
-    def make_control_header(
+    def call_action(
+        self,
+        service_name,
+        action,
+        body=None,
+        **kwargs
+    ):
+        """
+        Build and send a single job request with one action.
+
+        Returns the action response or raises an exception if the action response is an error.
+
+        Args:
+            action: string
+            body: dict
+            switches: list of ints
+            context: dict
+            correlation_id: string
+        Returns:
+            ActionResponse
+        """
+        action_request = ActionRequest(
+            action=action,
+            body=body or {},
+        )
+        return self.call_actions(
+            service_name,
+            [action_request],
+            **kwargs
+        ).actions[0]
+
+    def call_actions(self, service_name, actions, expansions=None, **kwargs):
+        """
+        Build and send a single job request with one or more actions.
+
+        Returns a list of action responses, one for each action, or raises an exception if any action response is an
+        error.
+
+        This method performs expansions if the Client is configured with an expansion converter.
+
+        Args:
+            actions: list of ActionRequest or dict
+            switches: list
+            context: dict
+            correlation_id: string
+            continue_on_error: bool
+            control_extra: dict
+        Returns:
+            JobResponse
+        """
+        request_id = self.send_request(service_name, actions, **kwargs)
+        # Dump everything from the generator. There should only be one response.
+        responses = list(self.get_all_responses(service_name))
+        response_id, response = responses[0]
+        if response_id != request_id:
+            raise Exception('Got response with ID {} for request with ID {}'.format(response_id, request_id))
+        if response.errors:
+            raise self.JobError(response.errors)
+        error_actions = [action for action in response.actions if action.errors]
+        if error_actions:
+            raise self.CallActionError(error_actions)
+
+        # Perform expansions
+        if expansions and hasattr(self, 'expansion_converter'):
+            # Initialize service request cache
+            exp_service_requests = {service_name: {}}
+
+            # Build initial list of objects to expand
+            objs_to_expand = []
+            for type_node in self.expansion_converter.dict_to_trees(expansions):
+                for action in response.actions:
+                    exp_objects = type_node.find_objects(action.body)
+                    objs_to_expand.extend(
+                        (exp_object, type_node.expansions)
+                        for exp_object in exp_objects
+                    )
+
+            # Loop until we have no outstanding requests or responses
+            while objs_to_expand or any(requests for requests in exp_service_requests.values()):
```

Comment:
It seemed that they weren't necessary. I tested this code path in both 2.7 and 3.5 and it worked as intended.